consul/consul.go

```diff
@@ -46,23 +47,23 @@ func (c *Consul) getServicesUsingProviderWithRetriesOnAgentFailure(provide Servi
 		if err != nil {
 			return nil, err
 		}
-		if services, err := provide(agent.Client); err != nil {
+		services, err := provide(agent.Client)
+		if err != nil {
 			log.WithError(err).WithField("Address", agent.IP).
 				Error("An error occurred getting services from Consul, retrying locally or with another agent")
 			if failures := agent.IncFailures(); agent != localAgent && failures > c.config.AgentFailuresTolerance {
 				log.WithField("Address", agent.IP).WithField("Failures", failures).
 					Warn("Removing agent due to too many failures")
 				c.agents.RemoveAgent(agent.IP)
 			}
-		} else {
-			agent.ClearFailures()
-			return services, nil
 		}
+		agent.ClearFailures()
```

Comment:
Missing else? You changed the logic here.

Reply:
Reverted